client/src/VideosList.js

```diff
@@ -0,0 +1,44 @@
+import React from "react";
+import VoteButton from "./VoteButton";
+
+
+const VideosList = ({videoData}) => {
+    //console.log(videoData)
+    const videoIdExtract = (url) =>{
+        var regExp =
+          /^.*((youtu.be\/)|(v\/)|(\/u\/\w\/)|(embed\/)|(watch\?))\??v?=?([^#&?]*).*/;
```

Comment:
Always use `let` or `const` instead of `var`.

Also, I can see that you have used a regular expression in this particular code. Can you please explain why have you used this regular expression? What is it doing and what are the benefits and disbenefits of using a regex here?

Reply:
Hi Ali
Thank you for making an effort looking at my work.
In this case I have used this regex to help extract the YouTube video ID from the url is I needed it as the iframe src value